.github/workflows/single-header.yml

```diff
@@ -0,0 +1,34 @@
+# Copyright 2023 Bernhard Manfred Gruber
+# SPDX-License-Identifier: MPL-2.0
+
+name: Publish amalgamated alpaka.hpp to single-header
+on:
+  push:
+    branches:
+      - develop
+
+jobs:
+  single-header:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: clone single-header
+      run: |
+        git clone -b single-header https://x-access-token:${{secrets.github_token}}@github.com/${{github.repository}}.git single-header
```

Comment:
Is the branch single-header only updated if we merge something to the develop branch or with any PR.
I am wondering because there is already a branch single header but I am not sure if @bernhardmgruber created the branch by hand or if this branch was auto-generated.

Reply:
OK my question is answered with line 8 https://github.com/alpaka-group/alpaka/pull/1956/files/e12634281d6207e29f1dd3841d6e933f1233a20f#diff-88ac08689a99a939e350a0c6fc02c3a5b93c0cf3637202361f6831ab3d0dc5aaR8

Only updated if we push to develop.

Reply:
Yes, as you have found out yourself, the `single-header` branch is updated on each push to the `develop` branch. I have created this branch manually and put an amalgamated `alpaka.hpp` there, to test manually whether the amalgamation script works. The name of the branch from which this PR is opened is called `single_header` (underbar vs. dash) and is an unrelated name.